### PR TITLE
Trip Plan Language | Transfer

### PR DESCRIPTION
**Pull request by thecristen**
#### Summary of changes
**Asana Ticket:** [Trip Plan Language | Transfer](https://app.asana.com/0/385363666817452/1179317340719423/f)

Changes "Depart onto Transfer" to either "Transfer" or "Depart" depending on whether the subsequent leg is transit or not, respectively.

---

Before getting review, please check the following:

* [x] Does frontend functionality render and work correctly in IE?
* [x] Have we load-tested any new pages or internal API endpoints that will receive significant traffic?
* [x] Are interactive elements accessible to screen readers?
* [x] Have you checked for tech debt you can address in the area you're working in?
* [x] If this change involves routes, does it work correctly with pertinent "unusual" routes such as the combined Green Line, Silver Line, Foxboro commuter rail, and single-direction bus routes like the 170?
* [x] Are the changes organized into self-contained commits with descriptive and well-formatted commit messages?

### CODE REVIEW

File: apps/site/lib/site/trip_plan/itinerary_row_list.ex [hunk 1]
```diff
@@ -49,9 +49,9 @@ defmodule Site.TripPlan.ItineraryRowList do
         ]
   defp get_rows(itinerary, deps, opts, alerts) do
     rows =
-      for leg <- itinerary do
+      for {leg, index} <- Enum.with_index(itinerary.legs) do
         leg
-        |> ItineraryRow.from_leg(deps)
+        |> ItineraryRow.from_leg(deps, Enum.at(itinerary.legs, index + 1))
```
Comment (alicia-mbta) [inline apps/site/lib/site/trip_plan/itinerary_row_list.ex:54]: Is there always going to be a `index + 1` element? or will we run into 'out of bounds' scenarios at some point?

Comment (thecristen) [reply apps/site/lib/site/trip_plan/itinerary_row_list.ex:54]: No and yes! `Enum.at` for an out of bounds index should return `nil` in that case.

**Comment by ingridpierre** [comment]
👍 lgtm!

**Comment by ingridpierre** [comment]
lgtm! 👍

**Pull request merged**